src/scripts/main.js

```diff
@@ -1,3 +1,162 @@
 'use strict';
 
 // write code here
+// Notifications
+
+const pushNotification = (posTop, posRight, title, description, type) => {
+  const notification = document.createElement('div');
+
+  notification.setAttribute('data-qa', 'notification');
+
+  const notificationTitle = notification.appendChild(
+    document.createElement('h1'),
+  );
+  const notificationDescription = notification.appendChild(
+    document.createElement('p'),
+  );
+
+  notificationDescription.textContent = description;
+  notificationTitle.textContent = title;
+
+  notification.classList.add(type);
+  notification.style.top = `${posTop}px`;
+  notification.style.right = `${posRight}px`;
+
+  document.body.appendChild(notification);
+
+  setTimeout(() => {
+    notification.style.display = 'none';
+  }, 2000);
+};
+
+// Sorting
+
+const sortButtons = document.querySelectorAll('th');
+const tableBody = document.querySelector('tbody');
+let ascOrder = true;
+const rows = Array.from(tableBody.rows);
+
+function sortTable(cellIndex, sortType) {
+  rows.sort((rowA, rowB) => {
+    let cellA = rowA.cells[cellIndex].textContent.trim();
+    let cellB = rowB.cells[cellIndex].textContent.trim();
+
+    if (cellIndex === 3 || cellIndex === 4) {
+      cellA = parseFloat(cellA.replace(/[$,]/g, '')) || 0;
+      cellB = parseFloat(cellB.replace(/[$,]/g, '')) || 0;
```

Comment:
The sorting logic assumes that columns with indices 3 and 4 are numeric and require special parsing. This approach may not be flexible if the table structure changes. Consider making the column index and type configurable or dynamically determined.